internal/dotc1z/sync_runs.go

```diff
@@ -351,3 +354,93 @@ func (c *C1File) EndSync(ctx context.Context) error {
 
 	return nil
 }
+
+func (c *C1File) Cleanup(ctx context.Context) error {
+	l := ctxzap.Extract(ctx)
+
+	if skipCleanup, _ := strconv.ParseBool(os.Getenv("BATON_SKIP_CLEANUP")); skipCleanup {
+		return nil
+	}
+
+	err := c.validateDb(ctx)
+	if err != nil {
+		return err
+	}
+
+	if c.currentSyncID != "" {
+		return nil
+	}
+
+	var ret []*syncRun
+
+	pageToken := ""
+	for {
+		runs, nextPageToken, err := c.ListSyncRuns(ctx, pageToken, 100)
+		if err != nil {
+			return err
+		}
+
+		for _, sr := range runs {
```

Comment:
Is this desired behavior? You could start and never complete multiple syncs right? Maybe make sure to keep the current 'active' one but all the others could go?